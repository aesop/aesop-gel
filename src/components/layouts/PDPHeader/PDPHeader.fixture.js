export const product = {
  name: 'Parsley Seed Facial Cleanser',
  field1Label: 'Suited to',
  field1Description: 'Normal, combination and troubled skin',
  field2Label: 'Skin feel',
  field2Description: 'Clarified, smooth and gently cleansed',
  variants: [
    {
      description:
        'A clarifying formulation that removes surface impurities and offers mild exfoliation from Lactic Acid. Ideal for maintaining immaculately clean skin in polluted urban environments.',
      image: {
        altText: 'here is a descriptive alt tag',
        sizes: [
          {
            viewport: 'LARGE',
            src:
              '/images/products/Skin_Amazing_Face_Cleanser_100mL_large.png 2x',
          },
          {
            viewport: 'MEDIUM',
            src:
              '/images/products/Skin_Amazing_Face_Cleanser_100mL_medium.png 2x',
          },
          {
            viewport: 'SMALL',
            src:
              '/images/products/Skin_Amazing_Face_Cleanser_100mL_small.png 2x',
          },
        ],
      },
      isDefault: true,
      keyIngredient: 'Licorice Root, Lactic Acid, Blackcurrant Seed',
      price: '$43.00',
      size: '100 mL',
      sku: 'SK45',
    },
  ],
};

export default {
  product,
};
