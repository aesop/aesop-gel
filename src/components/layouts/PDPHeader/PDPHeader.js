import React from 'react';
import cx from 'classnames';
import get from 'lodash/get';
import PDPDetailsHeader from '~/components/layouts/PDPDetailsHeader';
import PDPDetailsImage from '~/components/layouts/PDPDetailsImage';
import PDPDetailsSection from '~/components/layouts/PDPDetailsSection';
import PDPDetailsSectionFixture from '~/components/layouts/PDPDetailsSection/PDPDetailsSection.fixture';
import Props from './PDPHeader.types';
import {
  getDescription,
  getImage,
  getKeyIngredient,
  getPrice,
  getSku,
} from './PDPHeader.utils';
import { ProductContext } from '~/components/contexts/ProductContext/ProductContext';
import styles from './PDPHeader.module.css';

const PDPHeader: React.FunctionComponent<Props> = ({ classSet }) => {
  const product = React.useContext(ProductContext);

  const {
    field1Description,
    field1Label,
    field2Description,
    field2Label,
    ingredientText,
    name = '',
    variants = [],
  } = product;

  const defaultVariant = variants.find(variant => variant.isDefault);
  const [selectedOption, setSelectedOption] = React.useState(
    get(defaultVariant, 'sku', ''),
  );

  if (!product) {
    return null;
  }

  const classSet: string = cx(styles.base, className);
  const variantOptions = variants.map(({ size, sku }) => ({ size, sku }));

  const selectedVarient = variants.find(
    variant => variant.sku === selectedOption,
  );

  return (
    <form className={classSet}>
      <div className={styles.productWrapper}>
        <div className={styles.productImageWrapper}>
          <PDPDetailsImage
            id={getSku(selectedVarient)}
            image={getImage(selectedVarient)}
            onVariantChange={e => setSelectedOption(e.target.value)}
            selectedOption={selectedOption}
            variantOptions={variantOptions}
          />
        </div>
        <div className={styles.productDetailsWrapper}>
          <PDPDetailsHeader
            description={getDescription(selectedVarient)}
            productName={name}
          />
          <PDPDetailsSection
            copy={PDPDetailsSectionFixture.copy}
            field1Description={field1Description}
            field1Label={field1Label}
            field2Description={field2Description}
            field2Label={field2Label}
            ingredientText={ingredientText}
            keyIngredient={getKeyIngredient(selectedVarient)}
            name={name}
            price={getPrice(selectedVarient)}
            sku={getSku(selectedVarient)}
          />
        </div>
      </div>
    </form>
  );
};

export default PDPHeader;
