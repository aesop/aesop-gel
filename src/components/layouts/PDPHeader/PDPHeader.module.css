@import 'styles/variables/index';
@import 'styles/mixins/index';

.base {
  padding: var(--layout-sm-spacing);

  @media (--viewport-md) {
    padding: 84px 34px 59px;
  }

  @media (--viewport-lg) {
    display: flex;
    min-height: 85vh;
    flex-direction: column;
    padding: 84px 34px;
  }

  @media (--viewport-xl) {
    position: relative;
    padding: calc(var(--layout-logo-offset-lg) - 5px) var(--layout-lg-spacing)
      100px;
  }

  @media (--viewport-vertical) {
    height: 85vh;
    min-height: 500px;
    max-height: 1080px;
  }
}

.productWrapper {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  justify-content: flex-end;

  @media (--viewport-md-only) {
    flex-direction: column;
  }

  @media (--viewport-xl) {
    justify-content: space-between;
  }
}

.productImageWrapper {
  max-width: 100%;
  margin: 0 auto;

  @media (--viewport-lg) {
    @mixin grid-pct 7, 12;
    position: relative;
    display: flex;
    align-items: flex-end;
    justify-content: center;
    transform: translateX(30px);
  }

  @media (--viewport-xl) {
    @mixin grid-pct 8, 12;
    justify-content: flex-end;
  }
}

.productDetailsWrapper {
  width: 100%;

  @media (--viewport-md) {
    @mixin grid-pct 3, 6;
    flex-grow: 0;
    flex-shrink: 0;
  }

  @media (--viewport-md-only) {
    align-self: flex-end;
  }

  @media (--viewport-lg) {
    display: flex;
    @mixin grid-pct 5, 12;
    flex-direction: column;
    padding-right: 4.5%;
  }

  @media (--viewport-xl) {
    @mixin grid-pct 4, 12;
    max-width: 550px;
  }
}
