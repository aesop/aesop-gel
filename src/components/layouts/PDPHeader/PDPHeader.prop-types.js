import PropTypes from 'prop-types';

export const PROP_TYPES = {
  classname: PropTypes.string,
};

export default PROP_TYPES;
