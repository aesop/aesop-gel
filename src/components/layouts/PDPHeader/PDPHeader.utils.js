import { ProductVariant, ProductVariantImage } from '~/types/Product';

export const getDescription = (
  defaultVariant: ProductVariant | undefined,
): string => (defaultVariant ? defaultVariant.description : '');

export const getKeyIngredient = (
  defaultVariant: ProductVariant | undefined,
): string | undefined =>
  defaultVariant ? defaultVariant.keyIngredient : undefined;

export const getImage = (
  defaultVariant: ProductVariant | undefined,
): ProductVariantImage | undefined =>
  defaultVariant ? defaultVariant.image : undefined;

export const getPrice = (
  defaultVariant: ProductVariant | undefined,
): string | undefined => (defaultVariant ? defaultVariant.price : undefined);

export const getSku = (
  defaultVariant: ProductVariant | undefined,
): string | undefined => (defaultVariant ? defaultVariant.sku : undefined);

export default {
  getDescription,
  getKeyIngredient,
  getImage,
  getPrice,
  getSku,
};
