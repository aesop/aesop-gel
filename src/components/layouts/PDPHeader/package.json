{
  "name": "PDPHeader",
  "version": "0.0.0",
  "main": "PDPHeader.js"
}
