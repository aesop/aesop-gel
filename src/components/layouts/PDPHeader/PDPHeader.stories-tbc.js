import React from 'react';
import { ApolloProvider } from '@apollo/react-hooks';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import 'normalize.css';
import { client } from '~/lib/apollo/client';
import '~/styles/base.module.css';
import '~/styles/storybook.module.css';
import PDPHeader from './PDPHeader';
import { ProductContext } from '~/components/contexts/ProductContext/ProductContext';

const product = {
  name: text('name', 'Parsley Seed Facial Cleanser'),
  field1Label: text('field1Label', 'Suited to'),
  field1Description: text(
    'field1Description',
    'Normal, combination and troubled skin',
  ),
  field2Label: text('field2Label', 'Skin feel'),
  field2Description: text(
    'field2Description',
    'Clarified, smooth and gently cleansed',
  ),
  ingredientText: text(
    'ingredientText',
    'Water (Aqua), Kaolin, Aloe Barbadensis Leaf Juice, Bentonite, Alcohol Denat., Glycerin, Illite, Phenoxyethanol, Lavandula Angustifolia (Lavender) Oil, Oenothera Biennis (Evening Primrose) Oil, Rosa Canina Fruit Oil, Linalool, Ormenis Multicaulis Oil, Limonene, Carum Petroselinum (Parsley) Seed Oil, Geraniol',
  ),
  variants: [
    {
      description: text(
        'variants.description',
        'A clarifying formulation that removes surface impurities and offers mild exfoliation from Lactic Acid. Ideal for maintaining immaculately clean skin in polluted urban environments.',
      ),
      image: {
        altText: 'here is a descriptive alt tag',
        sizes: [
          {
            viewport: 'LARGE',
            src:
              '/images/products/Skin_Amazing_Face_Cleanser_100mL_large.png 2x',
          },
          {
            viewport: 'MEDIUM',
            src:
              '/images/products/Skin_Amazing_Face_Cleanser_100mL_medium.png 2x',
          },
          {
            viewport: 'SMALL',
            src:
              '/images/products/Skin_Amazing_Face_Cleanser_100mL_small.png 2x',
          },
        ],
      },
      isDefault: true,
      keyIngredient: text(
        'variants.keyIngredient',
        'Licorice Root, Lactic Acid, Blackcurrant Seed',
      ),
      price: text('variants.price', '$43.00'),
      size: text('variants.size', '100 mL'),
      sku: text('variants.sku', 'SK45'),
    },
  ],
};

storiesOf('Layouts.PDPHeader', module).add('Base component', () => (
  <ApolloProvider client={client}>
    <ProductContext.Provider value={product}>
      <PDPHeader />
    </ProductContext.Provider>
  </ApolloProvider>
));
