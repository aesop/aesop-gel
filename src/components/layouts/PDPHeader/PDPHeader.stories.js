import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import 'normalize.css';
import '~/styles/base.module.css';
import '~/styles/storybook.module.css';
import PDPHeader from './PDPHeader';
// import { ProductContext } from '~/components/contexts/ProductContext/ProductContext';

const product = {
  name: text('name', 'Parsley Seed Facial Cleanser'),
  field1Label: text('field1Label', 'Suited to'),
  field1Description: text(
    'field1Description',
    'Normal, combination and troubled skin',
  ),
  field2Label: text('field2Label', 'Skin feel'),
  field2Description: text(
    'field2Description',
    'Clarified, smooth and gently cleansed',
  ),
  ingredientText: text(
    'ingredientText',
    'Water (Aqua), Kaolin, Aloe Barbadensis Leaf Juice, Bentonite, Alcohol Denat., Glycerin, Illite, Phenoxyethanol, Lavandula Angustifolia (Lavender) Oil, Oenothera Biennis (Evening Primrose) Oil, Rosa Canina Fruit Oil, Linalool, Ormenis Multicaulis Oil, Limonene, Carum Petroselinum (Parsley) Seed Oil, Geraniol',
  ),
  variants: [
    {
      description: text(
        'variants.description',
        'A clarifying formulation that removes surface impurities and offers mild exfoliation from Lactic Acid. Ideal for maintaining immaculately clean skin in polluted urban environments.',
      ),
      image: {
        altText: 'here is a descriptive alt tag',
        sizes: [
          {
            viewport: 'LARGE',
            src:
              'https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ 2x',
          },
          {
            viewport: 'MEDIUM',
            src:
              'https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ 2x',
          },
          {
            viewport: 'SMALL',
            src:
              'https://www.aesop.com/medias/Aesop-SkinParsley-Seed-Facial-Cleanser-100mL-large.png?context=bWFzdGVyfGltYWdlc3wzNDYwMTh8aW1hZ2UvcG5nfGltYWdlcy9oYWMvaDIxLzg4MTUwMjM1ODczNTgucG5nfGFlYmNhMjRiM2Q2NWMzZTA1ZjljOTEyZmMxYmUyMTMyZDgyYjY4MGJlODU2OWEyNjkzYWQ0ZTBmMDY0NmQ1NjQ 2x',
          },
        ],
      },
      isDefault: true,
      keyIngredient: text(
        'variants.keyIngredient',
        'Licorice Root, Lactic Acid, Blackcurrant Seed',
      ),
      price: text('variants.price', '$43.00'),
      size: text('variants.size', '100 mL'),
      sku: text('variants.sku', 'SK45'),
    },
  ],
};

storiesOf('Layouts.PDPHeader', module).add('Base component', () => (
  <PDPHeader product={product} />
));
