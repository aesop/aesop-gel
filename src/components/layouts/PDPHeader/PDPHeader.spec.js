import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MockedProvider } from '@apollo/react-testing';
import renderer, { act } from 'react-test-renderer';
import waitForExpect from 'wait-for-expect';
import PDPHeader from './PDPHeader';
import PDPHeaderFixture from './PDPHeader.fixture';
import { ProductContext } from '~/components/contexts/ProductContext/ProductContext';

configure({ adapter: new Adapter() });

describe('<PDPHeader />', () => {
  it('should be defined', () => {
    expect(PDPHeader).toBeDefined();
  });

  it('renders base component correctly', async () => {
    const component = renderer.create(
      <ProductContext.Provider value={PDPHeaderFixture.product}>
        <MockedProvider mocks={[]}>
          <PDPHeader />
        </MockedProvider>
      </ProductContext.Provider>,
    );

    await act(async () => {
      await waitForExpect(() => {
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
