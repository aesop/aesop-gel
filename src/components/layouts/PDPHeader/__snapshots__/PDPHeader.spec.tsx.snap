// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PDPHeader /> renders base component correctly 1`] = `
<form
  className="base"
>
  <div
    className="productWrapper"
  >
    <div
      className="productImageWrapper"
    >
      <div
        className="base"
        id="product-SK45"
      >
        <picture
          className="picture base image"
        >
          <source
            media="(min-width: 1025px)"
            srcSet="https://acceptance.aesop.com/images/products/Skin_Amazing_Face_Cleanser_100mL_large.png 2x"
          />
          <source
            media="(min-width: 640px)"
            srcSet="https://acceptance.aesop.com/images/products/Skin_Amazing_Face_Cleanser_100mL_medium.png 2x"
          />
          <source
            media="(min-width: 0px)"
            srcSet="https://acceptance.aesop.com/images/products/Skin_Amazing_Face_Cleanser_100mL_small.png 2x"
          />
          <img
            alt="here is a descriptive alt tag"
          />
        </picture>
        <div
          className="singleVariantLabel"
        >
          100 mL
        </div>
      </div>
    </div>
    <div
      className="productDetailsWrapper"
    >
      <header
        className="base"
      >
        <h1
          className="base xLarge productTitle"
        >
          Parsley Seed Facial Cleanser
        </h1>
        <div
          className="productDesc"
        >
          <p>
            A clarifying formulation that removes surface impurities and offers mild exfoliation from Lactic Acid. Ideal for maintaining immaculately clean skin in polluted urban environments.
          </p>
          

        </div>
      </header>
      <div
        className="base"
      >
        <div
          className="large xLarge hidden"
        >
          <button
            className="base alternate block base action"
            data-test-ref="ADD_TO_CART_SMALL_CTA"
            disabled={false}
            onClick={[Function]}
            title="Add to cart button"
            type="button"
          >
            <span
              className="label"
            >
              <span>
                Added to your cart
              </span>
              <span>
                Add to your cart — $43.00
              </span>
            </span>
          </button>
        </div>
        <dl
          className="base"
        >
          <dt
            className="term"
          >
            Suited to
          </dt>
          <dd
            className="description"
          >
            Normal, combination and troubled skin
          </dd>
          <dt
            className="term"
          >
            Skin feel
          </dt>
          <dd
            className="description"
          >
            Clarified, smooth and gently cleansed
          </dd>
          <dt
            className="term"
          >
            <span>
              Key ingredients
            </span>
          </dt>
          <dd
            className="description"
          >
            Licorice Root, Lactic Acid, Blackcurrant Seed
          </dd>
        </dl>
        <div
          className="small medium hidden"
        >
          <button
            className="base alternate block base action"
            data-test-ref="ADD_TO_CART_LARGE_CTA"
            disabled={false}
            onClick={[Function]}
            title="Add to cart button"
            type="button"
          >
            <span
              className="label"
            >
              <span>
                Added to your cart
              </span>
              <span>
                Add to your cart — $43.00
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</form>
`;
